### ISSUE TITLE
Question: Deploying part of a workspace

### ISSUE THREAD

**Issue by james-rabbit** [yarnpkg/yarn#6494]
I have a monorepo that has a couple libraries and a bunch of services built on top of them.
With workspaces, it is convenient to develop these locally.
However, this makes it difficult to deploy services separately.

```
LibA depends on:
 - external deps
LibB depends on:
 - external deps
Service1 depends on:
 - LibA
 - external deps
Service2 depends on:
 - LibA
 - LibB
 - external deps
```

Normally I would copy LibA or LibB to the node_modules of the services that need them, zip up each service folder, and I can deploy those on to servers.

With workspaces, I can sort of make it work by merging the shared `node_modules` from the base directory with the ones in each service, but it includes a lot of dependencies that each service doesn't need. With many services, it really adds up and makes zipping and deploying unfeasible using workspaces.

Is there a better way to package up a service with all the dependencies it needs, without including the ones it does not need?

**Comment by james-rabbit** [comment]
Thanks for the links, I took some time to read through all of it.
I saw some script solutions in the linked issues, but they either were very project-specific or just copied every dependency to each workspace path.

The ideal solution would probably be a yarn command that does the normal workspace install, and then 'un-hoists' the required dependencies for each workspace by copying just the necessary files into each of their `node_modules` to make them work independently.

If that is outside the scope of yarn it might be doable in a script, if it's not too hard to parse the dependency structure of a yarn workspace setup.

**Comment by creatyvtype** [comment]
Having the same exact problem. We are using workspaces very nicely for packages, but now that we want to introduce our deployables, this is a blocker. Is there any proposed work on this? Will this be added?

**Comment by arcanis** [comment]
My solution to this in the [v2](https://github.com/yarnpkg/berry/) is to remove as much as possible from the install itself. Ideally you wouldn't care about installing the dependencies of one independent workspaces, because it wouldn't change anything in the install time.

In order to do that, the v2 stores the package archives within the project folder, and uses PnP to instruct Node to load the files directly from there. It means that right after being cloned, your project is ready to be used in any environment you would want to use it (bar any native package, which will still require a `yarn install` in order to get built).

I should write an article about it ... despite being relatively simple it's easy to overlook.

**Comment by merceyz** [comment]
Closing as fixed in v2 where you can use the plugin https://yarn.build/ or https://gitlab.com/Larry1123/yarn-contrib/-/tree/master/packages/plugin-production-install to do this

https://yarnpkg.com/getting-started/migration
